src/main/java/it/gov/pagopa/payhub/activities/activity/ingestionflow/debtposition/SynchronizeIngestedDebtPositionActivityImpl.java

```diff
@@ -0,0 +1,59 @@
+package it.gov.pagopa.payhub.activities.activity.ingestionflow.debtposition;
+
+import it.gov.pagopa.payhub.activities.connector.debtposition.DebtPositionService;
+import it.gov.pagopa.pu.debtposition.dto.generated.PagedDebtPositions;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Slf4j
+@Lazy
+@Service
+public class SynchronizeIngestedDebtPositionActivityImpl implements SynchronizeIngestedDebtPositionActivity {
+
+    private final DebtPositionService debtPositionService;
+    private final Integer pageSize;
+
+    private static final List<String> DEFAULT_ORDERING = List.of("debtPositionId,asc");
+
+    public SynchronizeIngestedDebtPositionActivityImpl(DebtPositionService debtPositionService,
+                                                       @Value("${query-limits.debt-positions.size}") Integer pageSize) {
+        this.debtPositionService = debtPositionService;
+        this.pageSize = pageSize;
+    }
+
+    @Override
+    public String synchronizeIngestedDebtPosition(Long ingestionFlowFileId) {
+        log.info("Synchronizing all debt positions related to ingestion flow file id {}", ingestionFlowFileId);
+
+        StringBuilder errors = new StringBuilder();
+
+        int currentPage = 0;
+        boolean hasMorePages = true;
+
+        while (hasMorePages) {
+            PagedDebtPositions pagedDebtPositions = debtPositionService.getDebtPositionsByIngestionFlowFileId(ingestionFlowFileId,
+                    currentPage,
+                    pageSize,
+                    DEFAULT_ORDERING);
+
```

Comment:
let's log page advance:
log.info("Synchronizing page {} of {} retrieved searching debt positions related to ingestionFlowFileId {} (totalElements {})",
currentPage, pagedDebtPositions.getTotalPages(), pagedDebtPositions.getTotalElements(), ingestionFlowFileId);